process/block/baseProcess.go

```diff
@@ -704,3 +740,154 @@ func (bp *baseProcessor) requestMissingFinalityAttestingHeaders(
 
 	return missingFinalityAttestingHeaders
 }
+
+func (bp *baseProcessor) isShardStuck(shardId uint32) bool {
```

Comment:
This method is used for checking if the notarized shard is stuck. It needs to access lastHdrForShard map which is constructed here in process/block and based on this map and current round of chronology it can decide if the given shard is stuck or not